src/passport-saml/saml.ts

```diff
@@ -459,7 +485,7 @@ class SAML {
       additionalParams[k] = optionsAdditionalParams[k];
     });
 
-    let optionsAdditionalParamsForThisOperation = {};
+    let optionsAdditionalParamsForThisOperation: Record<string, string> = {};
```

Comment:
I've noticed that you use this `Record<>` construct a lot in this very generic way. I'm not used to seeing it used this way. Instead, I've more accustomed to seeing `{[K: string]: string}`. Is there a reason this notation was chosen? It seems much more C#-y than JavaScript-y.